### PR TITLE
nixos/mysql: specify option types, add tests

### PR DESCRIPTION
**Pull request by florianjacob**
###### Motivation for this change
Prevent bugs, as mentioned in
https://github.com/NixOS/nixpkgs/pull/56720#issuecomment-471194925

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by florianjacob** [comment]
@GrahamcOfBorg test mysql

**Comment by florianjacob** [comment]
@GrahamcOfBorg test mysql

**Comment by florianjacob** [comment]
@GrahamcOfBorg test mysql

**Comment by florianjacob** [comment]
@GrahamcOfBorg test mysql

**Comment by aanderse** [comment]
PS. @ryantm have you had any thoughts on https://github.com/NixOS/nixpkgs/issues/50403? Any thoughts from you, @florianjacob?

File: nixos/modules/services/databases/mysql.nix [hunk 1]
```diff
@@ -134,6 +154,36 @@ in
...
+                The attribute values specfiy the permissions to grant.
+                You may specify one or multiple comma-separated SQL privileges here.
+
+                For more information on how to specify the target
+                and on which privileges exist, see the
+                <link xlink:href="https://mariadb.com/kb/en/library/grant/">GRANT syntax</link>.
+                The attributes are used as <code>GRANT ''${attrName} ON ''${attrValue}</code>.
+              '';
+              example = {
+                "database.*" = "ALL PRIVILEGES";
```
Comment (Infinisil) [inline nixos/modules/services/databases/mysql.nix]: This needs `example = literalExample ''...` to show up correcly formatted in the docs

Comment (florianjacob) [reply nixos/modules/services/databases/mysql.nix]: Tiny little missing " I failed to see. *squints at manual*

Also fixed the wrong indentation of the ensureUsers top-level literal example.

**Comment by aanderse** [comment]
@GrahamcOfBorg test mysql

**Pull request merged**